### ISSUE TITLE
Crash when merging overlapping monitors

### ISSUE THREAD

**Issue by jhyeon** [baskerville/bspwm#239]
I have an external monitor of the resolution 1920x1080 attached to my laptop.  The following commands crash bspwm:

    $ bspc config merge_overlapping_monitors true
    $ xrandr --output LVDS1 --mode 1366x768 --panning 1920x1080+0+0 --output VGA1 --mode 1920x1080 --panning 1920x1080+0+0

```
GNU gdb (Gentoo 7.9 vanilla) 7.9
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://bugs.gentoo.org/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from bspwm...done.
(gdb) r
Starting program: bspwm
update monitors
add monitor LVDS1 (0x41)
add desktop Desktop1
update motion recorder size: 1366x768
update monitors
update monitor LVDS1 (0x41)
update motion recorder size: 1366x768
update monitors
update monitor LVDS1 (0x41)
update motion recorder size: 1366x768
update monitors
update monitor LVDS1 (0x41)
update motion recorder size: 1366x768
add desktop L2
add desktop L3
add desktop L4
add desktop L5
add desktop L6
client message 2E09BB3 333
configure request 1000004
configure request 1000004
unmap notify 2E09BB3
unmap notify 2E09BB3
client message 2E0A2A5 333
configure request 1000004
configure request 1000004
unmap notify 2E0A2A5
unmap notify 2E0A2A5
adopt orphans
add pending rule 6
add pending rule 7
add pending rule 8
add pending rule 9
add pending rule 10
add pending rule 11
add pending rule 12
add pending rule 13
add pending rule 14
add pending rule 15
window show 1000004
remove pending rule 7
window show 1600003
remove pending rule 6
window show 100002E
remove pending rule 10
window show 1000025
remove pending rule 8
window show 100002B
remove pending rule 9
window show 1000028
remove pending rule 11
manage 2E0000C
insert node 2E0000C
arrange LVDS1 L1
stack 2E0000C
focus node 2E0000C
window show 2E0000C
remove pending rule 14
manage 36000A9
insert node 36000A9
floating 36000A9: true
update vacant state
stack 36000A9
arrange LVDS1 L1
stack 36000A9
focus node 36000A9
window show 36000A9
remove pending rule 15
configure request 1000004
configure request 2E0000C
Failed request: SetInputFocus, BadMatch: 56623273.
unmap notify 36000A9
unmanage 36000A9
remove node 36000A9
unlink node 36000A9
stack 2E0000C
focus node 2E0000C
arrange LVDS1 L1
unmap notify 36000A9
client message 2E0A2A5 333
configure request 1000004
manage 4200001
insert node 4200001
arrange LVDS1 L1
stack 4200001
focus node 4200001
window show 4200001
remove pending rule 12
manage 400007F
insert node 400007F
arrange LVDS1 L1
stack 400007F
focus node 400007F
window show 400007F
remove pending rule 13
configure request 1000004
configure request 1000004
configure request 1000004
configure request 2E0000C
client message 3200032 333
configure request 1000004
configure request 1000004
configure request 2E0000C
client message 40001CB 333
unmap notify 3200032
unmap notify 3200032
unmap notify 40001CB
unmap notify 40001CB
unmap notify 40001CB
configure request 1000004
unmap notify 2E0A2A5
unmap notify 2E0A2A5
grab pointer 1
stack 2E0000C
focus node 2E0000C
configure request 1000004
grab pointer 1
grab pointer 1
grab pointer 1
client message 2E09BB3 333
configure request 1000004
configure request 1000004
unmap notify 2E09BB3
unmap notify 2E09BB3
update monitors
arrange LVDS1 L1
update monitor LVDS1 (0x41)
add monitor VGA1 (0x42)
remove monitor VGA1 (0x42)
merge  into LVDS1
remove monitor  (0x42)
*** Error in `bspwm': double free or corruption (fasttop): 0x0000000000628200 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x770af)[0x7ffff6b5f0af]
/lib64/libc.so.6(+0x7c9ce)[0x7ffff6b649ce]
/lib64/libc.so.6(+0x7d6e6)[0x7ffff6b656e6]
/home/jhpark/hck/tile/bspwm/bspwm[0x405acd]
/home/jhpark/hck/tile/bspwm/bspwm[0x40683f]
/home/jhpark/hck/tile/bspwm/bspwm[0x40ba2f]
/home/jhpark/hck/tile/bspwm/bspwm[0x404252]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7ffff6b0cdc5]
/home/jhpark/hck/tile/bspwm/bspwm[0x403a79]
======= Memory map: ========
00400000-0041d000 r-xp 00000000 08:14 11933470                           /home/jhpark/hck/tile/bspwm/bspwm
0061c000-0061d000 r--p 0001c000 08:14 11933470                           /home/jhpark/hck/tile/bspwm/bspwm
0061d000-0061e000 rw-p 0001d000 08:14 11933470                           /home/jhpark/hck/tile/bspwm/bspwm
0061e000-00640000 rw-p 00000000 00:00 0                                  [heap]
7ffff64c8000-7ffff64dd000 r-xp 00000000 08:13 6312516                    /usr/lib64/gcc/x86_64-pc-linux-gnu/4.8.4/libgcc_s.so.1
7ffff64dd000-7ffff66dc000 ---p 00015000 08:13 6312516                    /usr/lib64/gcc/x86_64-pc-linux-gnu/4.8.4/libgcc_s.so.1
7ffff66dc000-7ffff66dd000 r--p 00014000 08:13 6312516                    /usr/lib64/gcc/x86_64-pc-linux-gnu/4.8.4/libgcc_s.so.1
7ffff66dd000-7ffff66de000 rw-p 00015000 08:13 6312516                    /usr/lib64/gcc/x86_64-pc-linux-gnu/4.8.4/libgcc_s.so.1
7ffff66de000-7ffff66e3000 r-xp 00000000 08:13 5779500                    /usr/lib64/libXdmcp.so.6.0.0
7ffff66e3000-7ffff68e2000 ---p 00005000 08:13 5779500                    /usr/lib64/libXdmcp.so.6.0.0
7ffff68e2000-7ffff68e3000 r--p 00004000 08:13 5779500                    /usr/lib64/libXdmcp.so.6.0.0
7ffff68e3000-7ffff68e4000 rw-p 00005000 08:13 5779500                    /usr/lib64/libXdmcp.so.6.0.0
7ffff68e4000-7ffff68e6000 r-xp 00000000 08:13 5779494                    /usr/lib64/libXau.so.6.0.0
7ffff68e6000-7ffff6ae6000 ---p 00002000 08:13 5779494                    /usr/lib64/libXau.so.6.0.0
7ffff6ae6000-7ffff6ae7000 r--p 00002000 08:13 5779494                    /usr/lib64/libXau.so.6.0.0
7ffff6ae7000-7ffff6ae8000 rw-p 00003000 08:13 5779494                    /usr/lib64/libXau.so.6.0.0
7ffff6ae8000-7ffff6c86000 r-xp 00000000 08:13 5677355                    /lib64/libc-2.19.so
7ffff6c86000-7ffff6e85000 ---p 0019e000 08:13 5677355                    /lib64/libc-2.19.so
7ffff6e85000-7ffff6e89000 r--p 0019d000 08:13 5677355                    /lib64/libc-2.19.so
7ffff6e89000-7ffff6e8b000 rw-p 001a1000 08:13 5677355                    /lib64/libc-2.19.so
7ffff6e8b000-7ffff6e8f000 rw-p 00000000 00:00 0
7ffff6e8f000-7ffff6e91000 r-xp 00000000 08:13 5653205                    /usr/lib64/libxcb-xinerama.so.0.0.0
7ffff6e91000-7ffff7090000 ---p 00002000 08:13 5653205                    /usr/lib64/libxcb-xinerama.so.0.0.0
7ffff7090000-7ffff7091000 r--p 00001000 08:13 5653205                    /usr/lib64/libxcb-xinerama.so.0.0.0
7ffff7091000-7ffff7092000 rw-p 00002000 08:13 5653205                    /usr/lib64/libxcb-xinerama.so.0.0.0
7ffff7092000-7ffff709f000 r-xp 00000000 08:13 5653145                    /usr/lib64/libxcb-randr.so.0.1.0
7ffff709f000-7ffff729e000 ---p 0000d000 08:13 5653145                    /usr/lib64/libxcb-randr.so.0.1.0
7ffff729e000-7ffff729f000 r--p 0000c000 08:13 5653145                    /usr/lib64/libxcb-randr.so.0.1.0
7ffff729f000-7ffff72a0000 rw-p 0000d000 08:13 5653145                    /usr/lib64/libxcb-randr.so.0.1.0
7ffff72a0000-7ffff72ac000 r-xp 00000000 08:13 6044678                    /usr/lib64/libxcb-ewmh.so.2.0.0
7ffff72ac000-7ffff74ac000 ---p 0000c000 08:13 6044678                    /usr/lib64/libxcb-ewmh.so.2.0.0
7ffff74ac000-7ffff74ad000 r--p 0000c000 08:13 6044678                    /usr/lib64/libxcb-ewmh.so.2.0.0
7ffff74ad000-7ffff74ae000 rw-p 0000d000 08:13 6044678                    /usr/lib64/libxcb-ewmh.so.2.0.0
7ffff74ae000-7ffff74b3000 r-xp 00000000 08:13 6044675                    /usr/lib64/libxcb-icccm.so.4.0.0
7ffff74b3000-7ffff76b2000 ---p 00005000 08:13 6044675                    /usr/lib64/libxcb-icccm.so.4.0.0
7ffff76b2000-7ffff76b3000 r--p 00004000 08:13 6044675                    /usr/lib64/libxcb-icccm.so.4.0.0
7ffff76b3000-7ffff76b4000 rw-p 00005000 08:13 6044675                    /usr/lib64/libxcb-icccm.so.4.0.0
7ffff76b4000-7ffff76b9000 r-xp 00000000 08:13 6169196                    /usr/lib64/libxcb-util.so.1.0.0
7ffff76b9000-7ffff78b8000 ---p 00005000 08:13 6169196                    /usr/lib64/libxcb-util.so.1.0.0
7ffff78b8000-7ffff78b9000 r--p 00004000 08:13 6169196                    /usr/lib64/libxcb-util.so.1.0.0
7ffff78b9000-7ffff78ba000 rw-p 00005000 08:13 6169196                    /usr/lib64/libxcb-util.so.1.0.0
7ffff78ba000-7ffff78da000 r-xp 00000000 08:13 5653072                    /usr/lib64/libxcb.so.1.1.0
7ffff78da000-7ffff7ad9000 ---p 00020000 08:13 5653072                    /usr/lib64/libxcb.so.1.1.0
7ffff7ad9000-7ffff7ada000 r--p 0001f000 08:13 5653072                    /usr/lib64/libxcb.so.1.1.0
7ffff7ada000-7ffff7adb000 rw-p 00020000 08:13 5653072                    /usr/lib64/libxcb.so.1.1.0
7ffff7adb000-7ffff7bdb000 r-xp 00000000 08:13 5679199                    /lib64/libm-2.19.so
7ffff7bdb000-7ffff7dda000 ---p 00100000 08:13 5679199                    /lib64/libm-2.19.so
7ffff7dda000-7ffff7ddb000 r--p 000ff000 08:13 5679199                    /lib64/libm-2.19.so
7ffff7ddb000-7ffff7ddc000 rw-p 00100000 08:13 5679199                    /lib64/libm-2.19.so
7ffff7ddc000-7ffff7dfd000 r-xp 00000000 08:13 5677838                    /lib64/ld-2.19.so
7ffff7fb1000-7ffff7fb7000 rw-p 00000000 00:00 0
7ffff7ff5000-7ffff7ff8000 rw-p 00000000 00:00 0
7ffff7ff8000-7ffff7ffa000 r--p 00000000 00:00 0                          [vvar]
7ffff7ffa000-7ffff7ffc000 r-xp 00000000 00:00 0                          [vdso]
7ffff7ffc000-7ffff7ffd000 r--p 00020000 08:13 5677838                    /lib64/ld-2.19.so
7ffff7ffd000-7ffff7ffe000 rw-p 00021000 08:13 5677838                    /lib64/ld-2.19.so
7ffff7ffe000-7ffff7fff000 rw-p 00000000 00:00 0
7ffffffdd000-7ffffffff000 rw-p 00000000 00:00 0                          [stack]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

Program received signal SIGABRT, Aborted.
0x00007ffff6b205c9 in raise () from /lib64/libc.so.6
(gdb) where
#0  0x00007ffff6b205c9 in raise () from /lib64/libc.so.6
#1  0x00007ffff6b21928 in abort () from /lib64/libc.so.6
#2  0x00007ffff6b5f0b4 in __libc_message () from /lib64/libc.so.6
#3  0x00007ffff6b649ce in malloc_printerr () from /lib64/libc.so.6
#4  0x00007ffff6b656e6 in _int_free () from /lib64/libc.so.6
#5  0x0000000000405acd in remove_monitor (m=0x628200) at monitor.c:185
#6  0x000000000040683f in update_monitors () at monitor.c:411
#7  0x000000000040ba2f in handle_event (evt=0x625aa0) at events.c:71
#8  0x0000000000404252 in main (argc=1, argv=0x7fffffffd4f8) at bspwm.c:181
(gdb) f 5
#5  0x0000000000405acd in remove_monitor (m=0x628200) at monitor.c:185
185             free(m);
(gdb) l
180                     mon = (last_mon == NULL ? (prev == NULL ? next : prev) : last_mon);
181                     if (mon != NULL && mon->desk != NULL)
182                             update_current();
183             }
184             xcb_destroy_window(dpy, m->root);
185             free(m);
186             num_monitors--;
187             put_status();
188     }
189
(gdb) q
```

**Comment by baskerville** [comment]
Should be fixed by a172aea.

**Comment by jhyeon** [comment]
It works.  Many thanks!
